delira/models/classification/ResNet18.py

```diff
@@ -17,53 +18,91 @@ def __init__(self, num_classes=None):
 
         self.conv2_1 = conv2d(filters=64, strides=1, kernel_size=3,
                               padding='same')
+        self.batchnorm2_1 = batchnorm2d(axis=1)
         self.conv2_2 = conv2d(filters=64, strides=1, kernel_size=3,
                               padding='same')
+        self.batchnorm2_2 = batchnorm2d(axis=1)
 
+        self.identity_3 = conv2d(filters=128, strides=2, kernel_size=1,
+                              padding='same')                              
+        self.batchnorm3 = batchnorm2d(axis=1)
         self.conv3_1 = conv2d(filters=128, strides=2, kernel_size=3,
                               padding='same')
+        self.batchnorm3_1 = batchnorm2d(axis=1)
         self.conv3_2 = conv2d(filters=128, strides=1, kernel_size=3,
                               padding='same')
+        self.batchnorm3_2 = batchnorm2d(axis=1)
 
+        self.identity_4 = conv2d(filters=256, strides=2, kernel_size=1,
+                              padding='same')                              
+        self.batchnorm4 = batchnorm2d(axis=1)
         self.conv4_1 = conv2d(filters=256, strides=2, kernel_size=3,
                               padding='same')
+        self.batchnorm4_1 = batchnorm2d(axis=1)
         self.conv4_2 = conv2d(filters=256, strides=1, kernel_size=3,
                               padding='same')
+        self.batchnorm4_2 = batchnorm2d(axis=1)
 
+        self.identity_5 = conv2d(filters=512, strides=2, kernel_size=1,
+                              padding='same')                              
+        self.batchnorm5 = batchnorm2d(axis=1)
         self.conv5_1 = conv2d(filters=512, strides=2, kernel_size=3,
                               padding='same')
+        self.batchnorm5_1 = batchnorm2d(axis=1)
         self.conv5_2 = conv2d(filters=512, strides=1, kernel_size=3,
                               padding='same')
+        self.batchnorm5_2 = batchnorm2d(axis=1)
 
-        self.gap = gap2d()
         self.dense1 = dense(1000)
         self.dense2 = dense(num_classes)
```

Comment:
This implementation would lead to a different network structure thanin torchvision. I think there should be a single fully connected layer with the correct number of classes (this would be consistent with torchvision).

Reply:
Although I think that the one in torchvision is not the original one state in the paper

Reply:
The original paper only refers to image net classification (if I'm not mistaken) and thus only covers the case with 1000 classes. In my opinion, it would make sense to keep the structure with one fully connected layer (as stated in the paper) and only reduce the number of output classes.
What are the differences between torchvision and the original paper?

Reply:
I think we should stick to the paper-version.